src/paho/mqtt/client.py

```diff
@@ -3740,7 +3968,18 @@ def _get_proxy(self):
         # None to indicate that the connection should be handled normally
         return None
 
-    def _create_socket_connection(self):
+    def _create_socket(self) -> SocketLike:
+        tcp_sock = self._create_socket_connection()
+        sock = self._create_ssl_socket_if_enable(tcp_sock)
```

Comment:
I'm not sure the `_if_enable` is a good idea - I'd just do

```suggestion
        if self._ssl:
            sock = self._ssl_wrap_socket(tcp_sock)
```
or similar? That would also simplify the return type of that function.